src/lib/provable/bigint.ts

```diff
@@ -141,6 +151,10 @@ function createProvableBigInt(modulus: bigint, config?: BigIntParameter) {
       return ProvableBigInt_.fromBigInt(BigInt(x));
     }
 
+    static toCanonical(x: ProvableBigInt_): ProvableBigInt_ {
+      return x.mul(ProvableBigInt_.one);
+    }
```

Comment:
this doesn't prove that the result is canonical 🤦🏻‍♂️ it witnesses an arbitrary $q, r$ such that $x * 1 = q * p + r$. nowhere does it prove that $r < p$. the prover could literally witness $q = 1$ and $r = x$.

note that multiplication also _shouldn't_ prove canoncal results. this method should have a check for that!
see how `ForeignField` does this:
https://github.com/o1-labs/o1js/blob/a6b65037a7ac1f1dbe647bb3767d00882cc3b02f/src/lib/provable/gadgets/foreign-field.ts#L429-L437